specifications/17-dependent-features.json

```diff
@@ -0,0 +1,738 @@
+{
+  "name": "17-dependent-features",
+  "state": {
+    "version": 1,
+    "features": [
+      {
+        "name": "parent.enabled",
+        "description": "Parent feature that is always enabled",
+        "enabled": true,
+        "strategies": [{
+          "name": "default"
+        }]
+      },
+      {
+        "name": "parent.disabled",
+        "description": "Parent feature that is always disabled",
+        "enabled": false,
+        "strategies": [{
+          "name": "default"
+        }]
+      },
+      {
+        "name": "parent.with.variant",
+        "description": "Parent feature with variant",
+        "enabled": true,
+        "strategies": [
+          {
+            "name": "flexibleRollout",
+            "parameters": {
+              "rollout": "100",
+              "stickiness": "default",
+              "groupId": "parent.with.variant"
+            },
+            "variants": [
+              {
+                "name": "parent.variant",
+                "weight": 1,
+                "payload": {
+                  "type": "string",
+                  "value": "variantValue"
+                }
+              }
+            ]
+          }
+        ]
+      },
+      {
+        "name": "parent.with.cycle",
+        "description": "Parent with cycle to child",
```

Comment:
Oh cool, new fear unlocked!

What's the actual data structure enforced by the server? Is this a graph or a tree? 

If it's a graph then cycle detection is much harder than this in the case where you have A -> B -> C -> A (please say it's not a graph, or if it is, a DAG enforced by the server)

If it's a tree, this shouldn't happen (but it's nice to be defensive!), hopefully that can be checked and blocked by the server

Reply:
There are no enforcements in the backend yet. But the rule will be simple: if your parent has any dependency it can't be your parent. This is our iteration 0.

Reply:
This is the line in SDK: 
```ts
if (this.repository.getToggle(parent.feature)?.dependencies?.length) {
        return false;
}
```

Reply:
> There are no enforcements in the backend yet. But the rule will be simple: if your parent has any dependency it can't be your parent. This is our iteration 0.

Awesome, that should handle cycles in either data structure. I think it's worth leaving in, but the only way this can trigger then is if you play silly buggers with your backup files